src/aiq/eval/dataset_handler/dataset_handler.py

```diff
@@ -81,6 +81,7 @@ def create_eval_item(row: pd.Series, structured: bool) -> EvalInputItem:
                 output_obj=row.get(self.generated_answer_key, "") if structured else "",
                 trajectory=row.get(self.trajectory_key, []) if structured else [],
                 expected_trajectory=row.get(self.expected_trajectory_key, []) if structured else [],
+                full_dataset_entry=row.to_dict() if self.dataset_config.pass_full_entry else None,
```

Comment:
Whats the downside to always including it? Memory? Minimizing output size?

Reply:
1. Yes on the memory increase, I am not entirely sure what else is maintained in a (non-ragas) dataset. But as I am also trying to allow dataset transformation in a different PR that may become less of a problem.
2.  I still need to maintain the other fields (input, expected_output) etc. in the EvalInputItem to avoid breaking backwards compatibility.